packages/block-editor/src/store/reducer.js

```diff
@@ -1842,6 +1842,55 @@ export function temporarilyEditingAsBlocks( state = '', action ) {
 	return state;
 }
 
+/**
+ * Reducer returning the current configuration for block inspector tabs.
+ *
+ * @param {Object} state  Current configuration state for block inspector tabs.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function blockInspectorTabs( state = {}, action ) {
+	switch ( action.type ) {
+		case 'ENABLE_BLOCK_INSPECTOR_TABS':
+			if ( ! action.blockName ) {
+				return { ...state, enabled: true };
+			}
+
+			return {
+				...state,
+				[ action.blockName ]: {
+					...state?.[ action.blockName ],
+					enabled: true,
+				},
+			};
+
+		case 'DISABLE_BLOCK_INSPECTOR_TABS':
+			if ( ! action.blockName ) {
+				return { ...state, enabled: false };
+			}
+
+			return {
+				...state,
+				[ action.blockName ]: {
+					...state?.[ action.blockName ],
+					enabled: false,
+				},
+			};
+
+		case 'SET_DEFAULT_BLOCK_INSPECTOR_TAB':
+			return {
+				...state,
+				[ action.blockName ]: {
+					...state?.[ action.blockName ],
+					defaultTab: action.defaultTab,
+				},
+			};
```

Comment:
The default tab could be a separate reducer, and it'd simplify the code quite a bit. `DISABLE_BLOCK_INSPECTOR_TABS` could be simplified to adding a block name to a unique array (so that the state is something like this `disabledInspectorTabs: [ 'core/freeform', ... ]`).

`SET_DEFAULT_BLOCK_INSPECTOR_TAB` could be storing a map of block names to tab names (`defaultInspectorTabs: { 'core/navigation': 'menu', ... }`).

Or you could use `combineReducers` to have nested state (e.g. with this shape - `inspectorTabs: { disabled: [ 'core/freeform' ], defaultTab: { 'core/navigation': 'menu' } }`

Reply:
Thanks for highlighting this.

> DISABLE_BLOCK_INSPECTOR_TABS could be simplified to adding a block name to a unique array (so that the state is something like this disabledInspectorTabs: [ 'core/freeform', ... ]).

I think we'll need to tweak the proposed structure a little further to allow for overriding the default display of tabs for _all blocks_.

Reply:
During the recent feature walkthrough for the block inspector tabs, it was deemed necessary so a plugin that may drastically overhaul the inspector controls can disable them across blocks. It was also decided that this shouldn't be a user preference option.